ngraph/python/src/ngraph/utils/tensor_iterator_types.py

```diff
@@ -0,0 +1,185 @@
+"""Helper classes for aggregating TensorIterator input/output desciptor attributes."""
+
+from typing import List
+
+from ngraph.impl import Node
+from ngraph.impl.op import Parameter
+
+
+class GraphBody(object):
+    """Class containing graph parameters and results."""
+
+    def __init__(
+        self,
+        parameters: List[Parameter],
+        results: List[Node],
+    ) -> None:
+        self.parameters = parameters
+        self.results = results
+
+    def serialize(self) -> dict:
+        return {
+            "parameters": self.parameters,
+            "results": self.results,
+        }
+
+
+class TensorIteratorInputDesc(object):
+    """Represents a generic input descriptor for TensorIterator operator."""
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+    ) -> None:
+        self.input_idx = input_idx
+        self.body_parameter_idx = body_parameter_idx
+
+    def serialize(self) -> dict:
+        return {
+            "input_idx": self.input_idx,
+            "body_parameter_idx": self.body_parameter_idx,
+        }
+
+
+class TensorIteratorSliceInputDesc(TensorIteratorInputDesc):
+    """Represents a TI graph body input formed from slicec of an TI input."""
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+        start: int,
+        stride: int,
+        part_size: int,
+        end: int,
+        axis: int,
+    ) -> None:
+        super(TensorIteratorSliceInputDesc, self).__init__(input_idx, body_parameter_idx)
+        self.start = start
+        self.stride = stride
+        self.part_size = part_size
+        self.end = end
+        self.axis = axis
+
+    def serialize(self) -> dict:
+        output = super(TensorIteratorSliceInputDesc, self).serialize()
+        output["start"] = self.start
+        output["stride"] = self.stride
+        output["part_size"] = self.part_size
+        output["end"] = self.end
+        output["axis"] = self.axis
+        return output
+
+
+class TensorIteratorMergedInputDesc(TensorIteratorInputDesc):
+    """Represents a TI graph body input with initial value in the first iteration.
+
+    Later on, this input value is computed inside graph body.
+    """
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+        body_value_idx: int,
+    ) -> None:
+        super(TensorIteratorMergedInputDesc, self).__init__(input_idx, body_parameter_idx)
+        self.body_value_idx = body_value_idx
+
+    def serialize(self) -> dict:
+        output = super(TensorIteratorMergedInputDesc, self).serialize()
+        output["body_value_idx"] = self.body_value_idx
+        return output
+
+
+class TensorIteratorInvariantInputDesc(TensorIteratorInputDesc):
+    """Represents a TI graph body input that has invariant value during iteration."""
+
+    def __init__(
+        self,
+        input_idx: int,
+        body_parameter_idx: int,
+    ) -> None:
```

Comment:
Why don't we put these signatures on one line? Try running `black --line-length=100` on this file.
```suggestion
    def __init__(self, input_idx: int, body_parameter_idx: int) -> None:
```